[pvr] change to new addon to kodi callback way

AlwinEsch opened the pull request:
## Description
This is the first major change to become multiple instances to work.

There are the PVR callbacks moved to the CPVRClient class. Reason is
to have this parts together and later with multiple instances to call
always right related CPVRClient class.

On addon is currently nothing needed to change, only a recompile to
match them.

<!--- Provide a general summary of your change in the Title above -->

<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo commented on xbmc/addons/PVRClient.cpp:
```diff
@@ -669,8 +700,8 @@ PVR_ERROR CPVRClient::GetEPGForChannel(const CPVRChannelPtr &channel, CPVREpg *e
   handle.dataAddress    = epg;
-  handle.dataIdentifier = bSaveInDb ? 1 : 0; // used by the callback method CAddonCallbacksPVR::PVRTransferEpgEntry()
-  PVR_ERROR retVal = m_struct.GetEpg(&handle,
+  handle.dataIdentifier = bSaveInDb ? 1 : 0; // used by the callback method CPVRClient::PVRTransferEpgEntry()
```
The method mentioned in the comment imo should read <code>CPVRClient::cb_transfer_epg_entry</code>

ksooo commented on xbmc/addons/PVRClient.cpp:
```diff
@@ -1565,29 +1589,432 @@ void CPVRClient::OnSystemSleep(void)
+{
+  int             iClientId;
+  unsigned int    iUniqueChannelId;
+  CPVREpgInfoTagPtr  event;
```
Could you please fix the spaces? I suggest to put only one blank between type and variable name.

ksooo reviewed (approved):
Looking good. Please don't forget to open PRs to bump API version and addon version and for changelog update for all PVR addons. Keep on rocking!

AlwinEsch commented:
jenkins build this please

AlwinEsch committed:
[pvr] change to new addon to kodi callback way

This is the first major change to become multiple instances to work.

There are the PVR callbacks moved to the CPVRClient class. Reason is
to have this parts together and later with multiple instances to call
always right related CPVRClient class.

On addon is currently nothing needed to change, only a recompile to
match them.

AlwinEsch commented:
jenkins build this please
